functional/keylime_tenant-signature-verification-options/test.sh

```diff
@@ -0,0 +1,120 @@
+        rlRun "limeWaitForRegistrar"
+        rlRun "limeStartAgent"
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+        # create allowlist and excludelist
+        limeCreateTestLists
+        # start simple http server to serve files
+        rlRun "python3 -m http.server 8000 &"
+        HTTP_PID=$!
+        # generate GPG key
+        rlRun "gpgconf --kill gpg-agent"
+        rlRun "export GNUPGHOME=${TmpDir}"
+        rlRun "cat >gpg.script <<EOF
+%echo Generating a basic OpenPGP key
+Key-Type: RSA
+Key-Length: 4096
+Subkey-Type: RSA
+Subkey-Length: 4096
+Name-Real: Joe Tester
+Name-Comment: with no passphrase
+Name-Email: joe@foo.bar
+Expire-Date: 0
+# Do a commit here, so that we can later print 'done' :-)
+%commit
+%echo done
+EOF"    
+        #generate gpg keys which will be genuine
+        rlRun "gpg --batch --pinentry-mode=loopback --passphrase '' --generate-key gpg.script"
+        rlRun "gpg --list-secret-keys"
+        rlRun "gpg --armor -o gpg-key-genuine.pub --export joe@foo.bar"
+        # sign our IMA key file
+        rlRun "gpg --detach-sign -o signature-gpg-genuine.sig ${limeIMAPublicKey}"
+        #generate fake gpg key
+        rlRun "gpg --batch --pinentry-mode=loopback --passphrase '' --generate-key gpg.script"
+        rlRun "gpg --armor -o gpg-key-fake.pub --export joe@foo.bar"
+        # sign our allowlist.txt and get fake signature
+        rlRun "gpg --detach-sign -o signature-gpg-fake.sig allowlist.txt"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Add keylime agent and validate IMA file signature verification key which is sign by GPG signature"
+        rlRun "gpg --verify signature-gpg-genuine.sig ${limeIMAPublicKey}"
+        rlRun "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --sign_verification_key ${limeIMAPublicKey} --signature-verification-key-sig signature-gpg-genuine.sig --signature-verification-key-sig-key gpg-key-genuine.pub -c add"
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Fail adding of keylime agent and validate IMA file signature verification key which is sign by GPG signature"
+        rlRun "gpg --verify signature-gpg-fake.sig allowlist.txt"
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --sign_verification_key ${limeIMAPublicKey} --signature-verification-key-sig signature-gpg-fake.sig --signature-verification-key-sig-key gpg-key-fake.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartTest "Add keylime agent and validate remote IMA file signature verification key which is sign by GPG signature"
```

Comment:
Please use something shorter, e.g. "Verify IMA key using a downloaded GPG signature and key"